### DISCUSSION TITLE
Any give me a example config for a reolink RLC-811A 4k camera .

### DISCUSSION THREAD

**Discussion by peter247** [blakeblackshear/frigate#2664] [General]
I'm looking for the rtmp and record   output_args / input_args ?
cam1:
    ffmpeg:
      inputs:
        - path: rtsp://******@192.168.10.101:554/h265Preview_01_main
          roles:
            - rtmp

[2022-01-17 22:52:14] frigate.video                  INFO    : cam1: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-17 22:52:14] frigate.video                  INFO    : cam1: ffmpeg process is not running. exiting capture thread...
[2022-01-17 22:52:31] watchdog.cam1                  ERROR   : FFMPEG process crashed unexpectedly for cam1.
[2022-01-17 22:52:31] watchdog.cam1                  ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-01-17 22:52:31] watchdog.cam1                  ERROR   : You may have invalid args defined for this camera.
[2022-01-17 22:52:31] ffmpeg.cam1.detect             ERROR   : [flv @ 0x559566cf0540] Video codec hevc not compatible with flv
[2022-01-17 22:52:31] ffmpeg.cam1.detect             ERROR   : Could not write header for output file #0 (incorrect codec parameters ?): Function not implemented
[2022-01-17 22:52:31] ffmpeg.cam1.detect             ERROR   : 
[2022-01-17 22:52:34] frigate.video                  INFO    : cam1: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
[2022-01-17 22:52:34] frigate.video                  INFO    : cam1: ffmpeg process is not running. exiting capture thread...
[2022-01-17 22:52:41] watchdog.cam1                  ERROR   : FFMPEG process crashed unexpectedly for cam1.
[2022-01-17 22:52:41] watchdog.cam1                  ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-01-17 22:52:41] watchdog.cam1                  ERROR   : You may have invalid args defined for this camera.
[2022-01-17 22:52:41] ffmpeg.cam1.detect             ERROR   : [flv @ 0x55e5fa409540] Video codec hevc not compatible with flv
[2022-01-17 22:52:41] ffmpeg.cam1.detect             ERROR   : Could not write header for output file #0 (incorrect codec parameters ?): Function not implemented

**Comment by micogg** [comment]
Hi Peter,

Let me start by saying I'm not an expert, just sharing what fixed my issue.  #2632 
Blake pointed out that there is camera specific recommendations for Reolink cameras.  It's for a different model, but may be worth trying.  

I originally was using the rtsp feed and getting the same errors as you.  I changed to the http feed based on the documentation and it resolved the issues.  The feed actually is a bit smoother now using the http feed.
There are some required arguments in the documentation you have to add for http.  One in particular that I added that wasn't in the documentation that seemed to be needed was an output arg.

output_args:
  rtmp: -c copy -f flv

Hope this gives you some direction while you wait for somebody with actual knowledge of this to assist!

**Reply by peter247** [reply]
Thank you, I got a RLC-811A and a RLC-812A 4k to try them out and they may finish up putting in a sd card hiding them somewhere and using their app.
I've tried motioneye and with the RLC-811A  I was watching at 25 fps without a problem , but the RLC-812A will work for a few seconds at 5 fps then looks to be resetting , Tried their app without problems .

isn't  rtmp: -c copy -f flv the default for rtmp ?

**Comment by nmsoares** [comment]
Hi guys.
Also struggling with the rlc-810a frigate config.
Can you share your configs ?

Thanks a million!

**Reply by peter247** [reply]
I don't think there is a way ? , I tried for a couple of days and finished up giving up, and just using the app for them.
so if you find any config which works let me know !!!
I THINK there is firmware which makes the camera 4mp and it uses h264 , but is the point of getting a 4k camera to be only to use it as a 4 mp camera.

**Comment by nmsoares** [comment]
@peter247 you don't need any firmware other than Reolink firmware to do that. I'm using the latest firmware from Reolink and you can specify a 4K resolution for the stream. I totally agree with your point...it make no sense to have an 8K working like a 4K camera.Anyway at least until there's not a "fix" we can have the system working in 4K :(

**Reply by peter247** [reply]
the problem is 4k in h265 and not h264 , so any camera with h265 you are going to have problems with .
motioneye works with h265 camera , so I guess the problem is with ffmpg that frigate uses to de/coding .

**Reply by blakeblackshear** [reply]
Frigate can do everything motioneye does with H265 streams. You just have to disable several features that motioneye doesn't have either.

**Reply by nmsoares** [reply]
@blakeblackshear Sorry for my ignorance. How would one configure Frigate then? Your help is very appreciated. Thanks!

**Reply by peter247** [reply]
+ 1 on that , it can works etc etc , but no one as given me a working config files or clues to get it working .
disable which features and how ?
as that song goes "I tried and i tried but i can't get no Satisfaction"

**Reply by blakeblackshear** [reply]
This basic config should work fine with H265 cameras. You can't use RTMP. Snapshots work fine. Recordings will only be viewable in Safari and Edge since they are the only browsers that play H265.

```yaml
cameras:
  cam1:
    ffmpeg:
      inputs:
        - path: rtsp://******@192.168.10.101:554/h265Preview_01_main
          roles:
            - detect
    rtmp:
      enabled: False
    detect:
      width: 1280 # <---- update for your camera's resolution
      height: 720 # <---- update for your camera's resolution
```

If you still have issues, please open an actual support issue. I prioritize support for those who use the actual support template: https://github.com/blakeblackshear/frigate/issues/new/choose

**Reply by peter247** [reply]
Thank you , I didn't include rtmp in the roles , but didn't know about the rtmp enable false line.
I can see the camera feed and no errors which is a first and a very good start .
I guess the detect width and height are the real resolution of the camera scream which is width: 2304 , height: 1296 minimum .
Thanks again from one happy bunny .

**Reply by blakeblackshear** [reply]
The width and height under detect can be lower than the real resolution of the camera and it will be resized before processing.

**Reply by nmsoares** [reply]
@peter247 can you share your config file? I want to test it out this evening. Thanks in advance

**Reply by peter247** [reply]
Sorry , one last thing , if I try to play a event back in a edge I get a  "Playback cannot continue. No available working or supported playlists." , is that normal ?, but I know I can just download the clip so not a big problem .

**Reply by justinbyoung** [reply]
The "Playback cannot continue. No available working or supported playlists." error is due to what you are viewing on having support for h265 playback in the browser. My linux laptop gives this error, but my Mac M1 laptop and my phone are both able to playback fine.

**Comment by peter247** [comment]
@nmsoares 

  cam1:
    ffmpeg:
      hwaccel_args:
        - -hwaccel
        -  vaapi
        - -hwaccel_device
        - /dev/dri/renderD128
        - -hwaccel_output_format
        - yuv420p
      input_args: ["-avoid_negative_ts","make_zero","-fflags","nobuffer+genpts+discardcorrupt","-rtsp_transport","tcp","-flags","low_delay","-strict","experimental","-analyzeduration","1000M","-probesize","1000M"]  
      #input_args: ["-avoid_negative_ts","make_zero","-fflags","nobuffer+genpts+discardcorrupt","-rtsp_transport","tcp"] 

      inputs:
        - path: rtsp://*******@192.168.10.101:554/h265Preview_01_main
          roles:
            - detect
            - record
      output_args:
         record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an 
    rtmp:
      enabled: False
    detect:                 
      width: 2304
      height: 1296
      fps: 5
    objects:
      track:
        - person
       # - car
        - dog
        - bicycle
        - cat
        - bird
        - horse
      filters:
        person:
          threshold: 0.7
        dog:
          threshold: 0.6
        cat:
          threshold: 0.6
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      retain:
        default: 5

    record:
      enabled: true
      retain_days: 0  
      events:
        retain:
          default: 10

**Comment by nmsoares** [comment]
Thanks @peter247 ! Will test it out tonight

**Reply by peter247** [reply]
@nmsoares , The important bit which I didn't know was - 
rtmp:
  enabled: False

**Comment by nmsoares** [comment]
@peter247 works perfectly! thanks @blakeblackshear and @peter247 .
Are you using for detecting that resolution  of 2304 by 1296 is all that resolution required that needed ? What do you gain?
Thanks!

**Reply by peter247** [reply]
2304 by 1296 is my native stream of the camera , using the full resolution is better for detecting small objects.
I had a trying at using all the resolutions and looking which was the clearest using the number plate test , and 2304 by 1296 looked to be best on my rlc-511 and the 4k camera aren't that stable at the full resolutions , but at 2304 by 1296 it is rock solid.

**Comment by asperkins** [comment]
Has anyone managed to get the 811a to work at max resolution? It works fine at the two lower resolutions, but not at the 3840x2160 setting.

**Reply by Gerjo** [reply]
The below config now works for me, record and detect. Reolink RLC-810A set to 3840x2160; 25 FPS; 6144 bitrate; 1x interframe space. Hopefully it works for your 811a too.

I've tried a million other permutations, nothing else worked for me at 4k, especially struggled to get anything from web2rtc as per recommendation in the manual for restreaming.

Running this on TrueNas Scale, with Frigate installed as "app". I wonder if things suddenly started to work after last weeks update; or a server restart in general makes it work (?!?!).

I use Safari to access the local frigate instance, that may or may not be a factor with h265 encoding. 

edit 1: I've added output_args to support audio recording.

```
mqtt:
  enabled: false
  
# https://docs.frigate.video/configuration/birdseye/
birdseye:
  enabled: True
  mode: continuous

record:
  enabled: true
  events: 
    pre_capture: 15
    post_capture: 15
    retain: 
      default: 10

# https://github.com/blakeblackshear/frigate/discussions/2664
cameras:
  cam1:
    ffmpeg:
      inputs:
        - path: rtsp://user:password@192.168.50.87:554/h265Preview_01_main
          roles:
            - record
            - detect
      output_args:
       record: preset-record-generic-audio-aac
    rtmp:
      enabled: False
    detect:
      width: 1280
      height: 720
```

**Reply by Gerjo** [reply]
Happy to say it also works for the Reolink WiFi doorbell and Reolink RLC-1224A (at 8k)

**Reply by yaneony** [reply]
Strange is, recorded video isn't working in Firefox (RLC-1224A).

**Reply by yaneony** [reply]
Tested everything I could find, but without success right now. Working fine on chrome, but not Firefox. And... Video is somehow laggy...

**Reply by NickM-27** [reply]
that's because firefox doesn't support h.265

**Reply by yaneony** [reply]
RLC-1224A stream is h264 afaik.

**Reply by NickM-27** [reply]
That is incorrect, all cameras with stream resolution > 5MP are h265

**Reply by yaneony** [reply]
Dang! You've right. Is there any way to make frigate output h264, so that it will work under firefox? I can see the stream, but recording aren't working.
